tempesta_fw/gfsm.c

```diff
@@ -236,7 +240,7 @@ tfw_gfsm_move(TfwGState *st, unsigned short state, const TfwFsmData *data)
 		 * rather than fixed priority levels to avoid spinning in vain.
 		 */
 		if (!(hooks[p] & mask))
```

Comment:
Just a refinement: as I understand, for now we should avoid to register hooks with priorities other then `TFW_GFSM_HOOK_PRIORITY_ANY`, since if we register e.g. three hooks for one state with priorities 0, 4, 5 - then hooks with 4 and 5 priorities will never be called.

Reply:
Correct. At the moment we must plan the hook priorities at programming stage. It's not good to walk over all unused priorities at hot path. We need to extend `tfw_gfsm_register_hook()` to build dense array of hooks.